descope/auth.py

```diff
@@ -105,18 +105,34 @@ def __init__(
                 self.public_keys = {kid: (pub_key, alg)}
 
     def _raise_rate_limit_exception(self, response):
-        resp = response.json()
-        raise RateLimitException(
-            resp.get("errorCode", HTTPStatus.TOO_MANY_REQUESTS),
-            ERROR_TYPE_API_RATE_LIMIT,
-            resp.get("errorDescription", ""),
-            resp.get("errorMessage", ""),
-            rate_limit_parameters={
-                API_RATE_LIMIT_RETRY_AFTER_HEADER: int(
-                    response.headers.get(API_RATE_LIMIT_RETRY_AFTER_HEADER, 0)
-                )
-            },
-        )
+        try:
+            resp = response.json()
+            raise RateLimitException(
+                resp.get("errorCode", HTTPStatus.TOO_MANY_REQUESTS),
+                ERROR_TYPE_API_RATE_LIMIT,
+                resp.get("errorDescription", ""),
+                resp.get("errorMessage", ""),
+                rate_limit_parameters={
+                    API_RATE_LIMIT_RETRY_AFTER_HEADER: self._parse_retry_after(
+                        response.headers
+                    )
+                },
+            )
+        except RateLimitException:
```

Comment:
why do we need this catch ? it will get here from the "try" section above (when parsed correctly)

Reply:
Otherwise it will go to the Expection, and raise the "genric" one

Reply:
oh right right, i missed the other exception..